salt/utils/event.py

```diff
@@ -873,6 +886,36 @@ def __del__(self):
         except:  # pylint: disable=W0702
             pass
 
+    def _verify_event_schema(self, tag, event):
+
+        if not self.opts['validate_event_schema']:
+            return True
+
+        if not HAS_JSONSCHEMA:
+            raise salt.exceptions.SaltException("Jsonschema not found, schema validation should be turned off")
+
+        schema = self._find_schema_by_tag(tag)
+        try:
+            jsonschema.validate(event, schema.serialize())
+        except jsonschema.exceptions.ValidationError as exc:
+            if self.opts['raise_on_invalid_schema']:
+                raise salt.exceptions.InvalidSchema(exc)
+            else:
+                log.error("Unable to validate schema for tag {0}".format(tag))
+
+    def _find_schema_by_tag(self, tag):
+        if len(SaltEvent.SCHEMAS) == 0:
+            for schema_mods in salt.loader.schemas(self.opts):
```

Comment:
Why are we calling the loader here? We should call call the loader when we init the class, I find this module level object to be very odd.

Reply:
You are correct, will fix this.